src/Trees/Helpers.hs

```diff
@@ -170,3 +173,13 @@ collectUniqueBinOpsInSynTree (Binary op l r) = nubOrd $ concat [
   [op],
   collectUniqueBinOpsInSynTree l,
   collectUniqueBinOpsInSynTree r]
+
+fillTreesRandomly :: [SynTree a ()] -> [b] -> Gen [SynTree a b]
```

Comment:
`fillTreesRandomly trees atoms = traverse (flip fillTreeRandomly atoms) trees` (and with other argument order for functions the flipping can be avoided)